.github/workflows/tg-demo-tests-impl.yaml

```diff
@@ -0,0 +1,49 @@
+name: "[internal] TG demo tests impl"
+
+on:
+  workflow_call:
+
+jobs:
+  tg-demo-tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        test-group: [
+          { name: "TG Llama3 demo tests", arch: wormhole_b0, model: llama3, timeout: 180, owner_id: U06F3ER8X9A}, # Stuti Raizada
+          { name: "TG Falcon7b demo tests", arch: wormhole_b0, model: falcon7b, timeout: 120, owner_id: U05RWH3QUPM}, # Salar Hosseini
+        ]
+    name: ${{ matrix.test-group.name }}
+    env:
+      TT_METAL_ENV: ${{ vars.TT_METAL_ENV }}
+      ARCH_NAME: ${{ matrix.test-group.arch }}
+      LOGURU_LEVEL: INFO
+      LD_LIBRARY_PATH: ${{ github.workspace }}/build/lib
+    runs-on:
+      - arch-wormhole_b0
+      - config-tg
+      - in-service
+      - bare-metal
+      - pipeline-functional
+    steps:
+      - uses: tenstorrent/tt-metal/.github/actions/checkout-with-submodule-lfs@main
+      - name: Set up dynamic env vars for build
+        run: |
+          echo "TT_METAL_HOME=$(pwd)" >> $GITHUB_ENV
+      - uses: actions/download-artifact@v4
+        with:
+          name: TTMetal_build_${{ matrix.test-group.arch }}
+      - name: Extract files
+        run: tar -xvf ttm_${{ matrix.test-group.arch }}.tar
+      - uses: ./.github/actions/install-python-deps
+      - name: Run demo regression tests
+        timeout-minutes: ${{ matrix.test-group.timeout }}
+        run: |
+          source ${{ github.workspace }}/python_env/bin/activate
+          cd $TT_METAL_HOME
+          export PYTHONPATH=$TT_METAL_HOME
+          ./tests/scripts/run_tests.sh --tt-arch wormhole_b0 --pipeline-type demos_tg_device --dispatch-mode "" --model ${{ matrix.test-group.model }}
```

Comment:
Can we try running in a docker container like `.github/workflows/fast-dispatch-build-and-unit-tests.yaml` so we can get started on this earlier?

Reply:
I couldn't get it to work.

[This Pipeline](https://github.com/tenstorrent/tt-metal/actions/runs/12314480754) was the longest I got, but everything crashes due to brisc build failed
This was the [commit used](https://github.com/tenstorrent/tt-metal/commit/02ae4b99125bd5b3ed6ce26e177de264d344bce9).

Since the TG pipelines were not using docker previously, we've pushed the PR like that. The Llama3 TG support is higher priority and has been delayed for over a week.

We can create a new smaller issue to tackle the proper usage of docker instances for TG pipelines in the future.